clang/lib/CIR/Dialect/Transforms/StdHelpers.h

```diff
@@ -29,6 +29,50 @@ namespace cir {
 
 bool isStdArrayType(mlir::Type t);
 
+// Recognizes a standard function represented by `StdFuncID` with arguments
+// count equal to `NumArgs`, and raise it to an instance of `TargetOp`. Using
+// IDs is a workaround, as we can't pass string literals to template arguments
+// in C++17.
+template <typename TargetOp> class StdRecognizer {
+private:
+  // Reserved for template specialization.
+  static bool checkArguments(mlir::ValueRange) { return true; }
```

Comment:
Looks like this function is a customization point. It might be more appropriate to move this argument checker to a dedicated template class:

```cpp
template <typename TargetOp> struct StdRecognizerArgsChecker {
  static bool check(mlir::ValueRange args) { return true; }
};
```

Otherwise specializations would have to re-implement the `raise` function.